packages/mongodb-artifact-generator/src/commands/generateReleaseNotes.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  createConfiguredAction,
+  withConfig,
+  withConfigOptions,
+} from "../withConfig";
+import { createCommand } from "../createCommand";
+import { makeRunLogger, type RunLogger } from "../runlogger";
+import { getReleaseArtifacts } from "../release-notes/getReleaseArtifacts";
+import { summarizeReleaseArtifacts } from "../release-notes/summarizeReleaseArtifacts";
+import {
+  ReleaseArtifact,
+  releaseArtifactShortMetadata,
+} from "../release-notes/projects";
+import { promises as fs } from "fs";
+import YAML from "yaml";
+import { ReleaseInfo } from "../release-notes/ReleaseInfo";
+
+let logger: RunLogger;
+
+type GenerateReleaseNotesCommandArgs = {
+  runId?: string;
+  releaseInfo: string;
+};
+
+export default createCommand<GenerateReleaseNotesCommandArgs>({
+  command: "generateReleaseNotes",
+  builder(args) {
+    return withConfigOptions(args)
+      .option("runId", {
+        type: "string",
+        demandOption: false,
+        description: "A (hopefully unique) name for the run.",
+      })
+      .option("releaseInfo", {
+        type: "string",
+        demandOption: true,
+        description:
+          "A path to a YAML file with release information. This file should contain the following keys: `github`, `jira`.",
+      });
+  },
+  async handler(args) {
+    logger = makeRunLogger({
+      topic: "GenerateReleaseNotes",
+      runId: args.runId,
+    });
+    logger.logInfo(`Running command with args: ${JSON.stringify(args)}`);
+    try {
+      const result = await withConfig(action, args);
+      logger.logInfo(`Success`);
+      return result;
+    } finally {
+      await logger.flushArtifacts();
+      await logger.flushLogs();
```

Comment:
just curious, what do these do?

Reply:
The logger keeps log lines & artifacts in memory until you flush them.

Flush logs => save to the logs file in the run directory
Flush artifacts => save each artifact to a file in the run directory